d8/d8batch/d8batch.module

```diff
@@ -0,0 +1,170 @@
+<?php
+
+/**
+ * @file
+ * Contains d8batch.module.
+ */
+
+use Drupal\Core\Entity\EntityStorageException;
+use Drupal\Core\Routing\RouteMatchInterface;
+use Symfony\Component\DomCrawler\Crawler;
+use Drupal\node\Entity\Node;
+use Drupal\Core\Url;
+
+/**
+ * Implements hook_help().
+ */
+function d8batch_help($route_name, RouteMatchInterface $route_match) {
+  switch ($route_name) {
+    // Main module help for the d8batch module.
+    case 'help.page.d8batch':
+      $output = '';
+      $output .= '<h3>' . t('About') . '</h3>';
+      $output .= '<p>' . t('Batch API demonstration in Drupal 8.') . '</p>';
+      return $output;
+
+    default:
+  }
+}
+
+/**
+ * Implements callback_batch_operation().
+ *
+ * @param string $url
+ *   Feed source URL.
+ */
+function _d8batch_batch_operation($url, array $context) {
+  try {
+    $url = Url::fromUri($url);
+    if (!d8batch_check_if_feed_source_already_exists($url)) {
+      try {
+        $client = new \GuzzleHttp\Client();
+        $response = $client->get($url->getUri());
+        if ($response->getStatusCode() == 200) {
+          // If title can not be extracted, then use the URL as title.
+          if (!($title = _d8batch_get_title_of_feed_source($response->getBody()
+            ->getContents()))
+          ) {
```

Comment:
This syntax looks really weird to me. Can you point me to an example in core which uses this same syntax?